### PR TITLE
feat!: add service_externalips option, and add default for autoscaling location_policy

### PR DESCRIPTION
**Pull request by jpetrucciani**
This PR adds the ability to set `service_external_ips_config` on the underlying cluster upon creation, which otherwise can (in my experience) take up to 20-30 minutes to enable independently after the cluster settles after initial creation. I've also added a default value for autoscaling `location_policy` for now, as this is something that appears to have a default on clusters created with 1.24+, but appears to be removed after a second plan+apply with this module.

I added the changes to the templates in `./autogen/` and ran `make build` in order to generate the changes across the repo

I've been testing cluster creation with these new settings already via my fork (and it appears to be working well! shaves a good amount of time off of our new cluster creation process), but would love to see it included in the official module if possible! Please let me know if there are any changes you'd like, or things I should take into account!

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@jpetrucciani
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -202,6 +202,10 @@ resource "google_container_cluster" "primary" {
     }
   }
 
+  service_external_ips_config {
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:205]: Can we make this a dynamic block if enabled?

File: autogen/main/variables.tf.tmpl [hunk 2]
```diff
@@ -96,6 +96,12 @@ variable "http_load_balancing" {
   default     = true
 }
 
+variable "service_externalips" {
```
Comment (bharathkkb) [inline autogen/main/variables.tf.tmpl:99]:
```suggestion
variable "service_external_ips" {
```

File: autogen/main/cluster.tf.tmpl [hunk 3]
```diff
@@ -616,8 +620,9 @@ resource "google_container_node_pool" "windows_pools" {
   dynamic "autoscaling" {
     for_each = lookup(each.value, "autoscaling", true) ? [each.value] : []
     content {
-      min_node_count = lookup(autoscaling.value, "min_count", 1)
-      max_node_count = lookup(autoscaling.value, "max_count", 100)
+      min_node_count  = lookup(autoscaling.value, "min_count", 1)
+      max_node_count  = lookup(autoscaling.value, "max_count", 100)
+      location_policy = "BALANCED"
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:625]: This seems like an unrelated change?

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @jpetrucciani 
Looks like this was added in 4.35 so we need to bump the minimum version in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/84fbb9a465a0e304910519d444f2fea40869464b/autogen/main/versions.tf.tmpl#L27

https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4350-september-6-2022

File: autogen/main/cluster.tf.tmpl (hunk 3 above)
Comment (jpetrucciani) [reply autogen/main/cluster.tf.tmpl:625]: Unrelated to external ips yes, but this was something I've seen happen in clusters as of 1.24 - they get this default value from the provider, but this module will remove it on the next plan + apply. I can separate into second PR (and make it a var) if you'd like!

File: autogen/main/variables.tf.tmpl (hunk 2 above)
Comment (jpetrucciani) [reply autogen/main/variables.tf.tmpl:99]: sure, sounds good - I will make some changes locally and update!

File: autogen/main/cluster.tf.tmpl (hunk 1 above)
Comment (jpetrucciani) [reply autogen/main/cluster.tf.tmpl:205]: Sure! let me include in the changes I'll make locally

**Comment by jpetrucciani** [comment]
made a new branch to represent just the main feature #1441

**Comment by comment-bot-dev** [comment]
@jpetrucciani
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.
